### PR TITLE
WindowServer+MouseSettings: Add cursor themes

### PR DESCRIPTION
**Pull request by sppmacd**
This PR adds:
* [X] Mouse theme support in WindowServer (Config + IPC calls)
* [x] Theme selector in MouseSettings
* [x] Dark theme
* [x] Update WidgetGallery to use new cursor path
* [x] Dark theme (32x32)
* [x] Persist theme when rebooting

### CODE REVIEW

File: Userland/Applications/MouseSettings/SchemeWidget.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    m_cursors_tableview->set_column_headers_visible(false);
+    m_cursors_tableview->set_highlight_key_column(false);
+
+    auto mouse_cursor_model = MouseCursorModel::create();
+    auto sorting_proxy_model = GUI::SortingProxyModel::create(mouse_cursor_model);
+    sorting_proxy_model->set_sort_role(GUI::ModelRole::Display);
+
+    m_cursors_tableview->set_model(sorting_proxy_model);
+    m_cursors_tableview->set_key_column_and_sort_order(MouseCursorModel::Column::Name, GUI::SortOrder::Ascending);
+    m_cursors_tableview->set_column_width(0, 25);
```
Comment (sppmacd) [inline Userland/Applications/MouseSettings/SchemeWidget.cpp:110]: This doesn't work for some reason:
image

Comment (sin-ack) [reply Userland/Applications/MouseSettings/SchemeWidget.cpp:110]: The wait cursor is expanding the table width. Animated cursors are stored as a single .png file with all the frames, so you will need to perform some modification before being able to use the animated icon bitmaps. Would be cool to have it animated in the table view though. :^)

Comment (sppmacd) [reply Userland/Applications/MouseSettings/SchemeWidget.cpp:110]: This works in WidgetGallery. I'll just parse the name as WindowServer does.

Comment (sppmacd) [reply Userland/Applications/MouseSettings/SchemeWidget.cpp:117]: I fixed that.

**Comment by awesomekling** [comment]
Please call them "Cursor Themes" instead of "Cursor Schemes"

**Comment by sppmacd** [comment]
> Please call them "Cursor Themes" instead of "Cursor Schemes"

Ok, this name is a lot better. "Cursor Schemes" is what Windows calls it.

**Comment by sppmacd** [comment]
I'll wait until all PRs about persistent model indices are merged.

**Pull request merged**